common/djangoapps/student/roles.py

```diff
@@ -136,7 +136,7 @@ class GlobalStaff(AccessRole):
     The global staff role
     """
     def has_user(self, user):
-        return bool(user and user.is_staff)
+        return bool(user and (user.is_superuser or user.is_staff))
```

Comment:
Conceptually, "global staff" and "super user" are two different roles. I don't think that this PR is helpful, because it says that "superusers have the global staff role", which I don't think is true.

There are two root problems and IMHO we should try to fix one of the root problems if we're going to be fixing something here:
1. The django `is_staff` flag is meant to indicate whether the user has access to the django admin. It is _not_ meant to be used for other things like indicating some openedx-specific "global staff" role that grants all kinds of other permissions. A new `is_global_staff` flag should be created to separate this role from the "has django admin access" role. (This is very similar to how the `is_active` flag is supposed to enabled/disable accounts, but is instead used to track email activation, which causes a number of bugs and requires workarounds throughout the system.)
2. Somewhere, the code to check "can this user access Studio" is implemented wrong. It should only be checking for a _permission_, and any permissions check will always return `True` for superusers. But instead, it's hard-coded to check for specific roles like "global staff", which is why the superuser role is seemingly not granting the permission.

CC @hsinkoff